### ISSUE TITLE
Odd Build Problems

### ISSUE THREAD

**Issue by NWPlayer123** [Steveice10/FBI#334]
Okay, so I'm trying to build from source and for some reason every time I install it, it gives me a black screen, I made sure everything was straight from git but it doesn't matter what I do. I tried compiling both from latest commit and 2.4.8 and both gave the same problem, only the cia provided works and I'm not sure what to do. Are you by any chance using older tools and some other piece broke it? Trying to narrow down if it's FBI or a dependency. (Also I'm using devkitARM r46)

**Comment by Steveice10** [comment]
I'm using devkitARM r46, and assuming you're using ctrulib 1.2.0 and the latest git commit of citro3d, we should have the same dependency versions as well. Just did a clean build myself and it seems to be working fine.

**Comment by NWPlayer123** [comment]
okay, so I found out the problem and it's really dumb. so I got it working with stock ctrulib 1.2.1 like you did, originally I was building from smea's repo which deprecated GPUCMD_Finalize(). Since citro3d uses it in source/base.c on line 275, it was giving me an error so I removed it to get it to build, thinking that deprecated means it didn't need to be called but nOPE
so I installed ctrulib 1.2.1 straight from the devkitPro Updater and citro3d compiled fine with it, and after I installed it, it works fine.

tl;dr use ctrulib 1.2.1 with citro3d from the repo for now when installing both libraries, since citro3d/FBI need GPUCMD_Finalize to display anything on startup.

**Comment by urherenow** [comment]
Sorry to chime in on a closed topic, but did you by chance try building this with the latest ctrulib and the "next" branch of citro3d? Did that give a black screen? I built it that way, but I haven't touched my 3ds since I got my hands on a switch, so I haven't actually tested it...

**Comment by NWPlayer123** [comment]
@urherenow yess that works perfectly, we're all good now
@Steveice10 so now that I know that, I'd edit the README to change it to do this, compile latest ctrulib and then the "next" branch of citro3d
git clone https://github.com/smealum/ctrulib.git && cd ctrulib/libctru && make && make install && cd ../..
git clone -b next https://github.com/fincs/citro3d.git && cd citro3d && make && make install && cd ..

**Comment by Steveice10** [comment]
I'm not officially using the next branch of citro3d. Waiting for it to stabilize and become mainstream.
